src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/ValueContentAnalysis/ValueContentAbstractValue.cs

```diff
@@ -154,7 +154,32 @@ protected override void ComputeHashCodeParts(ref RoslynHashCode hashCode)
         protected override bool ComputeEqualsByHashCodeParts(CacheBasedEquatable<ValueContentAbstractValue> obj)
```

Comment:
❓ Can you review the complexity of changing the internal hash code of `null` to a non-zero constant? I see two potential ways to handle this:

1. Make a `NullSentinel` type with a singleton instance that returns a known hash code, and use that to represent null
2. Change the way RoslynHashCode computes the hash code for null instances to default to a constant that is not zero

Option (2) would be simpler, but relies on all calls to `GetHashCode()` (and all uses of `EqualityComparer<T>`) to have the same understanding of the hash code for null. Option (1) is more complex, but forces the correct result.

Reply:
None of the options are viable as I found it's easy to still get collisions even when null isn't involved. See https://github.com/dotnet/roslyn-analyzers/pull/6876#discussion_r1299392581